src/TrayIcon.cpp

```diff
@@ -114,12 +115,33 @@ TrayIcon::TrayIcon(const QString &filename, QWindow *parent)
 
     menu->addAction(viewAction_);
     menu->addAction(quitAction_);
+
+    QString toolTip = QLatin1String("nheko");
+    QString profile = UserSettings::instance()->profile();
+    if (!profile.isEmpty()) {
+        toolTip.append(tr("\nProfile: %1").arg(profile));
+    }
+
+    setToolTip(toolTip);
 }
 
 void
 TrayIcon::setUnreadCount(int count)
 {
     qGuiApp->setBadgeNumber(count);
+    if (count != previousCount) {
+        QString currentToolTip = toolTip();
+        const QString before   = tr("\n%n unread message(s)", "", previousCount);
```

Comment:
I think you should simplify this and just always rewrite the whole message. Doing a text replacement isn't more efficient and harder to read (imo).